src/rdb/rdb_tx.c

```diff
@@ -388,6 +388,21 @@ rdb_tx_append(struct rdb_tx *tx, struct rdb_tx_op *op, bool is_critical)
 	return 0;
 }
 
+void
+rdb_tx_discard(struct rdb_tx *tx)
+{
+	D_ASSERT(!(tx->dt_flags & RDB_TX_LOCAL));
+	D_ASSERTF(
+	    (tx->dt_entry != NULL && tx->dt_entry_cap > 0 && tx->dt_entry_len <= tx->dt_entry_cap),
+	    "entry=%p cap=%zu len=%zu\n", tx->dt_entry, tx->dt_entry_cap, tx->dt_entry_len);
```

Comment:
I think this is a bit too strict; we could allow discarding all updates in a TX that has no update, so that when we get a "final" or "definitive" result, we may call `rdb_tx_discard` without worrying whether there has been at least one update in the TX (I imagine such cases with no updates might be numerous.)

Reply:
OK, I'll reuse the same assertion as in rdb_tx_append()